### PR TITLE
Use common agent config package for parsing process-config

### PR DESCRIPTION
**Pull request by sunhay**
This PR is just part 1 of a series. First step is updating the processes YAML config.

- Use the new common config utility package: `github.com/DataDog/datadog-agent/pkg/config`
- Tests were updated to use resource files for example YAML configuration files.

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -250,6 +251,12 @@ func TestDDAgentConfigWithNewOpts(t *testing.T) {
 }
 
 func TestDDAgentConfigBothVersions(t *testing.T) {
+	origcfg := config.Datadog
+	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
```
Comment (sfluor) [inline config/config_test.go]: since `config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))` is often used should we have a helper for it ?

Comment (sunhay) [reply config/config_test.go]: It's just tests + a one-liner so I didn't feel too strongly.

Comment (sfluor) [reply config/config_test.go]: For the `origcfg` we could just store it as a global variable in the test file and have a `resetGlobalConfig` helper

Comment (sunhay) [reply config/config_test.go]: Makes sense - I'll do that

File: config/config_test.go [hunk 2]
```diff
@@ -709,38 +607,28 @@ func TestGetProxySettings(t *testing.T) {
...
-		// Fake the os.Setenv("DD_SITE", tc.site)
-		ddconfig.Datadog.Set("site", tc.site)
-		os.Setenv("DD_PROCESS_AGENT_URL", tc.ddURL)
 
-		agentConfig, err := NewAgentConfig(nil, &YamlAgentConfig{}, nil)
-		assert.NoError(err)
-		assert.Equal(tc.expected, agentConfig.APIEndpoints[0].Endpoint.Hostname())
-	}
+	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	agentConfig, err := NewAgentConfig(nil, &NetworkAgentConfig{}, "./testdata/TestEnvSiteConfig.yaml")
```
Comment (sfluor) [inline config/config_test.go]: I believe this test was to test the behavior when we had `site` set in env variables too as well as the `process_dd_url` variable in the configuration, could we add this back ?

Comment (sunhay) [reply config/config_test.go]: Worth doing for a single test, but seems messy to do it X times. I'll add it back for a single case

Comment (sfluor) [reply config/config_test.go]: 👍

File: config/yaml_config.go [hunk 3]
```diff
@@ -263,25 +95,149 @@ func mergeNetworkYamlConfig(agentConf *AgentConfig, networkConf *YamlAgentConfig
...
+	a.setCheckInterval(ns, "container_realtime", "rtcontainer")
+	a.setCheckInterval(ns, "process", "process")
+	a.setCheckInterval(ns, "process_realtime", "rtprocess")
+	a.setCheckInterval(ns, "connections", "connections")
+
+	// A list of regex patterns that will exclude a process if matched.
+	for _, b := range config.Datadog.GetStringSlice(key(ns, "blacklist_patterns")) {
+		r, err := regexp.Compile(b)
+		if err != nil {
+			log.Warnf("Ignoring invalid blacklist pattern: %s", b)
```
Comment (sfluor) [inline config/yaml_config.go]: should we `continue` ?

Comment (sunhay) [reply config/yaml_config.go]: Good point - yes.

File: config/yaml_config.go [hunk 4]
```diff
@@ -263,25 +95,149 @@ func mergeNetworkYamlConfig(agentConf *AgentConfig, networkConf *YamlAgentConfig
...
+	a.Transport = ddutil.CreateHTTPTransport()
+
 	return nil
 }
+
+func (a *AgentConfig) setCheckInterval(ns, check, checkKey string) {
+	if interval := config.Datadog.GetInt(key(ns, "intervals", check)); interval != 0 {
+		log.Infof("Overriding container check interval to %ds", interval)
+		a.CheckIntervals[checkKey] = time.Duration(interval) * time.Second
+	}
```
Comment (sfluor) [inline config/yaml_config.go]: should we log something if the interval is 0 ?

Comment (sunhay) [reply config/yaml_config.go]: No - default would be zero so we'd always see that log message

**Pull request merged**